src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -277,21 +305,33 @@ internal AggregateToken ParseAggregate()
 
             this.lexer.NextToken();
 
-            return new AggregateToken(statements);
+            return statements;
         }
 
-        internal AggregateExpressionToken ParseAggregateExpression()
+        internal AggregateTokenBase ParseAggregateExpression()
         {
             // expression
-            var expression = this.ParseExpression();
+            QueryToken expression = ParseLogicalOr();
+
+            if (this.lexer.CurrentToken.Kind == ExpressionTokenKind.OpenParen)
+            {
+                // When there's a parenthesis after the expression we have a entity set aggregation.
+                // The syntax is the same as the aggregate expression itself, so we recurse on ParseAggregateExpressions.
+                this.aggregateExpressionParents.Push(expression);
+                var statements = ParseAggregateExpressions();
+                this.aggregateExpressionParents.Pop();
+
+                // TODO: since we ignore parsing errors in functions, we need to provide a nice error message that comprehend both cases.

```

Comment:
>// TODO: since we ignore parsing errors in functions, we need to provide a nice error message that comprehend both cases. [](start = 16, length = 121)

Is there an Issue for this? how will we know to come back to it?